### ISSUE TITLE
`Generator::lib_name` and `with_lib_name` should take `AsRef<str>` instead of `AsRef<String>`

### ISSUE THREAD

**Issue by ghost** [godot-rust/project-utils#8]
Neither `String` or `&String` actually implement `AsRef<String>`. To work around that using standard types I had to wrap the string with a smart pointer: `.lib_name(Box::new("mylibrary".to_string()))`.

**Comment by karroffel** [comment]
Oh that's an oversight, thanks for reporting! :+1: I'll change it right away!

**Comment by karroffel** [comment]
The type seems to be `AsRef<str>` which both `String`, `&String` and `&str` should implement, no?

https://github.com/godot-rust/project-utils/blob/dd8aaa1ce54f3e36aa5fec0b9b05851ac62d99f4/src/generate.rs#L71-L82

**Comment by karroffel** [comment]
It seems like that's an issue on the latest `crates.io` version but the git version should work!

**Comment by ghost** [comment]
Yes, I was using the crates version. I'll switch to the git version for now. Sorry for not checking it first.
